src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -450,6 +462,9 @@ void RayletBasedActorScheduler::HandleWorkerLeaseReply(
           << actor->GetActorID()
           << " has been already cancelled. The response will be ignored. Job id = "
           << actor->GetActorID().JobId();
+      if (actor->GetState() == rpc::ActorTableData::DEAD) {
+        kill_worker();
```

Comment:
Can you write a comment what's the condition of this, and why we call kill worker?